libcfnet/client_code.c

```diff
@@ -154,6 +154,17 @@ int TLSConnect(ConnectionInfo *conn_info, bool trust_server,
 
     const char *key_hash = KeyPrintableHash(conn_info->remote_key);
 
+    // If restrict_key is defined, check if the key is there
+    if (restrict_keys) {
```

Comment:
`restrict_keys != NULL` and please put the curly brackets on separate lines. (I'd personally write it the same way as you did, but the coding guidelines in CFEngine say otherwise).